Temporary back port from the GCC 10 branch until GCC 10.3 releases.

commit 3b777ab44049e06dc0825f86a8310673cdb6498c
Author: Francois-Xavier Coudert <fxcoudert@gcc.gnu.org>
Date:   Mon Dec 21 21:06:59 2020 +0000

    aarch64 : Remove the use of a C++11 brace-init-list.
    
    Instead of using a brace-init-list, set up the member values
    manually.
    
    gcc/
    
            * config/aarch64/aarch64-builtins.c
            (aarch64_init_memtag_builtins): Manually initialize instead
            of using a C++11 brace-init-list.

--- UTC
--- gcc/config/aarch64/aarch64-builtins.c
+++ gcc/config/aarch64/aarch64-builtins.c
@@ -1223,8 +1223,9 @@ aarch64_init_memtag_builtins (void)
     = aarch64_general_add_builtin ("__builtin_aarch64_memtag_"#N, \
 				   T, AARCH64_MEMTAG_BUILTIN_##F); \
   aarch64_memtag_builtin_data[AARCH64_MEMTAG_BUILTIN_##F - \
-			      AARCH64_MEMTAG_BUILTIN_START - 1] = \
-				{T, CODE_FOR_##I};
+			      AARCH64_MEMTAG_BUILTIN_START - 1].ftype = T; \
+  aarch64_memtag_builtin_data[AARCH64_MEMTAG_BUILTIN_##F - \
+			      AARCH64_MEMTAG_BUILTIN_START - 1].icode = CODE_FOR_##I;
 
   fntype = build_function_type_list (ptr_type_node, ptr_type_node,
 				     uint64_type_node, NULL);
